wp-content/themes/goonj-crm/functions.php

```diff
@@ -59,10 +59,30 @@ function goonj_login_form_validation_errors( $string ) {
         return '<p class="error">Login failed: Invalid username or password.</p>';
     }
 
+    if ( isset( $_REQUEST['password'] ) && $_REQUEST['password'] === 'true'  ) {
+        return '<p class="password-message">You can now log-in using your email address and the password you\'ve already set.</p>';
+    }
+
     return $string;
 }
 
 add_action('login_form_rp', 'goonj_custom_reset_password_form');
 function goonj_custom_reset_password_form() {
     get_template_part('templates/password-reset');
 }
+
+add_action( 'validate_password_reset', 'goonj_custom_password_reset_redirection', 10, 2 );
+function goonj_custom_password_reset_redirection( $errors, $user ) {
+    if ( $errors->has_errors() ) {
+        return;
+    }
+
+    if ( isset( $_POST['pass1'] ) && ! empty( $_POST['pass1'] ) ) {
+        reset_password( $user, $_POST['pass1'] );
+        $rp_cookie = 'wp-resetpass-' . COOKIEHASH;
+        $rp_path = preg_replace( '|https?://[^/]+|i', '', get_permalink( get_page_by_path( 'wp-login.php' ) ) );
```

Comment:
Here we are transforming the full URL of the WordPress login page file into just the path component.

For example, if the URL is `https://example.com/wp-login.php`
- The regular expression https?://[^/]+ matches https://example.com.
- The preg_replace function replaces https://example.com with an empty string, leaving just /wp-login.php.

So, after this line executes, $rp_path will contain `/wp-login.php`.
Leaving only the path part (/wp-login.php), this is necessary because the setcookie function requires the path to be in this format, not a full URL, to set the cookie correctly that's why I used this.

On more research, the expression can be simplified using $_SERVER['REQUEST_URI'] to directly obtain the path .`/wp-login.php`.This approach avoids the need for get_page_by_path(), get_permalink(), and preg_replace().